Xamarin.Forms.Platform.Android/ButtonLayoutManager.cs

```diff
@@ -0,0 +1,271 @@
+
+						var measuredTextWidth = view.Paint.MeasureText(buttonText, 0, buttonText.Length());
+						var textWidth = Math.Min((int)measuredTextWidth, view.Layout.Width);
+						var contentsWidth = ViewCompat.GetPaddingStart(view) + iconWidth + view.CompoundDrawablePadding + textWidth + ViewCompat.GetPaddingEnd(view);
+
+						var newLeft = (view.MeasuredWidth - contentsWidth) / 2;
+						if (_element.ContentLayout.Position == Button.ButtonContentLayout.ImagePosition.Right)
+							newLeft = -newLeft;
+						if (ViewCompat.GetLayoutDirection(view) == ViewCompat.LayoutDirectionRtl)
+							newLeft = -newLeft;
+
+						_drawableBounds.Set(newLeft, _drawableBounds.Top, newLeft + iconWidth, _drawableBounds.Bottom);
+						drawable.Bounds = _drawableBounds;
+					}
+				}
+			}
+		}
+
+		public void OnViewAttachedToWindow(AView attachedView)
+		{
+			Update();
+		}
+
+		public void OnViewDetachedFromWindow(AView detachedView)
+		{
+		}
+
+		public void Update()
+		{
+			if (!UpdateTextAndImage())
+				UpdateImage();
+			UpdatePadding();
+		}
+
+		void OnElementChanged(object sender, VisualElementChangedEventArgs e)
+		{
+			if (_element != null)
+			{
+				_element.PropertyChanged -= OnElementPropertyChanged;
+				_element = null;
+			}
+
+			if (e.NewElement is Button button)
+			{
+				_element = button;
+				_element.PropertyChanged += OnElementPropertyChanged;
+			}
+
+			Update();
+		}
+
+		void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			if (_disposed || _renderer == null || _element == null)
+				return;
+
+			if (e.PropertyName == Button.PaddingProperty.PropertyName)
+				UpdatePadding();
+			else if (e.PropertyName == Button.ImageProperty.PropertyName || e.PropertyName == Button.ContentLayoutProperty.PropertyName)
+				UpdateImage();
+			else if (e.PropertyName == Button.TextProperty.PropertyName || e.PropertyName == VisualElement.IsVisibleProperty.PropertyName)
+				UpdateTextAndImage();
+			else if (e.PropertyName == Button.BorderWidthProperty.PropertyName && _borderAdjustsPadding)
+				_element.InvalidateMeasureNonVirtual(InvalidationTrigger.MeasureChanged);
+		}
+
+		void UpdatePadding()
+		{
+			if (_disposed || _renderer == null || _element == null)
+				return;
+
+			AppCompatButton view = View;
+			if (view == null)
+				return;
+
+			if (!_defaultPaddingPix.HasValue)
+				_defaultPaddingPix = new Thickness(view.PaddingLeft, view.PaddingTop, view.PaddingRight, view.PaddingBottom);
+
+			var padding = _element.Padding;
+
+			var adjustment = 0.0;
+			if (_borderAdjustsPadding && _element is IBorderElement borderElement && borderElement.IsBorderWidthSet() && borderElement.BorderWidth != borderElement.BorderWidthDefaultValue)
+				adjustment = borderElement.BorderWidth;
+
+			var defaultPadding = _preserveInitialPadding && _defaultPaddingPix.HasValue
+				? _defaultPaddingPix.Value
+				: new Thickness();
+
+			view.SetPadding(
+				(int)(Context.ToPixels(padding.Left + adjustment) + defaultPadding.Left),
```

Comment:
To disable the default padding, you can set the `preserveInitialPadding` parameter to false: https://github.com/xamarin/Xamarin.Forms/pull/4967/files/737d1a953fa48f8248c829dc599a88e2ebf39c90#diff-08a5663cd61b123664c0ac0fd19c3255R31

Also, I think that the default padding is actually 0... If you look at the images, it appears to be the fact that the spacing is changed. I think there was a bug in the previous versions that ignored the padding: https://imgur.com/f3PSDDA

The default spacing is 10, but it looks the same as iOS - which ignored the spacing. So in 3.3, the spacing was observed. If I am not mistaken, Android has a default spacing between the text/image, so we may have to see what that is.

Not sure exactly, but I must test the spacing and see if the sizes are correct because I believe the default padding for appcompat/fast is actually 0, so this is basically a no-op. Material has a default padding, so we have to keep that - otherwise the image/text hangs over the edge of the button.